zebra-chain/src/serialization/date_time.rs

```diff
@@ -34,6 +34,21 @@ impl DateTime32 {
             .try_into()
             .expect("unexpected out of range chrono::DateTime")
     }
+
+    /// If `earlier` is less than or equal to `self`, returns the number of
+    /// seconds elapsed between `earlier` and `self`.
+    ///
+    /// Otherwise, returns `None`.
+    pub fn checked_duration_since(&self, earlier: DateTime32) -> Option<u32> {
+        self.timestamp.checked_sub(earlier.timestamp)
+    }
+
+    /// Returns the number of seconds elapsed since this time.
+    ///
+    /// If this time is in the future, returns `None`.
+    pub fn elapsed(&self) -> Option<u32> {
+        DateTime32::now().checked_duration_since(*self)
+    }
```

Comment:
These functions are based on parts of the `tokio::time::Instant` API, but without any panics:
https://docs.rs/tokio/1.6.2/tokio/time/struct.Instant.html#method.checked_duration_since
https://docs.rs/tokio/1.6.2/tokio/time/struct.Instant.html#method.elapsed